e2e/e2e_test.go

```diff
@@ -92,34 +92,43 @@ func Test_http_body(t *testing.T) {
 	stdErr, kill := startEnvoy(t, 8001)
 	defer kill()
 
-	for _, tc := range []struct {
-		op, expBody string
-	}{
-		{op: "append", expBody: `[original body][this is appended body]`},
-		{op: "prepend", expBody: `[this is prepended body][original body]`},
-		{op: "replace", expBody: `[this is replaced body]`},
-		// Should fall back to to the replace.
-		{op: "invalid", expBody: `[this is replaced body]`},
+	for _, mode := range []string{
+		"request",
+		"response",
 	} {
-		t.Run(tc.op, func(t *testing.T) {
-			require.Eventually(t, func() bool {
-				req, err := http.NewRequest("PUT", "http://localhost:18000/anything",
-					bytes.NewBuffer([]byte(`[original body]`)))
-				require.NoError(t, err)
-				req.Header.Add("buffer-operation", tc.op)
-				res, err := http.DefaultClient.Do(req)
-				if err != nil {
-					return false
-				}
-				defer res.Body.Close()
-				body, err := io.ReadAll(res.Body)
-				require.NoError(t, err)
-				return string(body) == tc.expBody &&
-					checkMessage(stdErr.String(), []string{
-						`original request body: [original body]`},
-						[]string{"failed to"},
-					) && checkMessage(string(body), []string{tc.expBody}, nil)
-			}, 5*time.Second, 500*time.Millisecond, stdErr.String())
+		t.Run(mode, func(t *testing.T) {
+			for _, tc := range []struct {
+				op, expBody string
+			}{
+				{op: "append", expBody: `[original body][this is appended body]`},
+				{op: "prepend", expBody: `[this is prepended body][original body]`},
+				{op: "replace", expBody: `[this is replaced body]`},
+				// Should fall back to to the replace.
+				{op: "invalid", expBody: `[this is replaced body]`},
+			} {
+				t.Run(tc.op, func(t *testing.T) {
```

Comment:
while you are here :bow:
```suggestion
				tc := tc
				t.Run(tc.op, func(t *testing.T) {				
```